clang/test/PCH/cxx1z-decomposition.cpp

```diff
@@ -22,7 +22,7 @@ constexpr int foo(Q &&q) {
   return a * 10 + b;
 }
 
-auto [noinit]; // expected-error{{decomposition declaration '[noinit]' requires an initializer}}
```

Comment:
This diagnostic here doesn't really seem like an improvement here?  Also, it reads pretty awkwardly to the point I don't have a suggestion on how to make it sensible  seeming.

The printing of the actual decomposition declaration was also nice, so its a shame to lose that.

Reply:
Does `decomposition declaration '[noinit]' requires an initializer; got ';'` sound better?